job_executor/model/metadata.py

```diff
@@ -310,7 +271,7 @@ def patch(self, other: "Variable") -> "Variable":
         if self.format is not None:
             patched.update({"format": self.format})
         if centralized_variable_definition:
-            patched.update({"keyType": self.key_type.dict()})
+            patched.update({"keyType": self.key_type.model_dump()})
```

Comment:
Let's use the aliases, and exclude nones here too for the sake of consistency 🙏🏻